src/features/groups/Details/index.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useCallback, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+import { PageWrapper } from 'components/Containers';
+import Header from 'components/Header';
+import { ResolveGroupRoute, ROUTES } from 'constants/routes';
+import { useLazyGetTransactionGroupByIdQuery } from '../api';
+import { ReactComponent as BackIconSVG } from 'assets/images/back-icon.svg';
+import { ReactComponent as EditIconSVG } from 'assets/images/edit-icon.svg';
+import BottomNavigation from 'components/BottomNavigation';
+import TransactionRow from './TransactionRow';
+import { ContentWrapper, DateText, HalfEllipse, NoTransactionsText, TransactionListContainer } from './index.styled';
+import { datesAreOnSameDay } from 'helpers/dateUtil';
+
+import { selectedGroupSelector } from '../slice';
+import Loader from 'components/Loader';
+
+
+const GroupDetails = () => {
+  const { t } = useTranslation('groups');
+  const navigate = useNavigate();
+  const { groupId } = useParams();
+
+  const [getGroup, { isLoading }] = useLazyGetTransactionGroupByIdQuery();
+  const group = useSelector(selectedGroupSelector);
+
+  const handleBackAction = useCallback(() => {
+    navigate(ROUTES.GROUPS.ROOT, { replace: true });
+  }, [navigate]);
+
+  const handleEditAction = useCallback(() => {
+    navigate(ResolveGroupRoute(`${ROUTES.GROUPS.EDIT}/${groupId}`));
+  }, [navigate, groupId]);
+
+  const createDateString = (date: Date) => {
+    if (date.getFullYear() === new Date().getFullYear()) {
+      return date.toLocaleDateString("en-GB", {day: 'numeric', month: 'long' });
+    }
+    else {
+      return date.toLocaleDateString("en-GB", {day: 'numeric', month: 'long', year: 'numeric' });
+    }
+  };
```

Comment:
Move it outside of the component.
Because it doesn't depend on some component internal data. And every time the component renders, it also recreates this function

Reply:
Done in fd1678776ebe70e333bc036eb07f0e36477a6142